quil-cli/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::Context;
+use clap::{Parser, Subcommand, ValueEnum};
+use quil_rs::{expression::Expression, quil::Quil, Program};
+use std::str::FromStr;
+
+#[derive(Parser, Debug)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand, Clone, Debug)]
+pub enum Command {
+    /// Parse a Quil program or expression
+    Parse {
+        #[arg(short = 't', long = "type")]
+        input_type: Option<InputType>,
```

Comment:
Same concept, but more direct, imo.

```suggestion
        #[arg(short = 't', long = "type", default_value_t)]
        input_type: InputType,
```